### PR TITLE
Adding isDeprecated hook definition

### PR DESCRIPTION
**Pull request by AMBudnik**
### Context

This pull request covers fixes for issues listed in #7400 and a small fix for #7399. All the issue are related to one of the following scenarios

* typos in parameters
* missing return types
* incorrect behavior

For more details visit #7400 

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7399
2. #7400

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable.d.ts [hunk 1]
```diff
@@ -1826,17 +1826,17 @@ declare namespace Handsontable {
       afterSelectionByProp?: (row: number, prop: string, row2: number, prop2: string, preventScrolling: { value: boolean }, selectionLayerLevel: number) => void;
       afterSelectionEnd?: (row: number, column: number, row2: number, column2: number, selectionLayerLevel: number) => void;
       afterSelectionEndByProp?: (row: number, prop: string, row2: number, prop2: string, selectionLayerLevel: number) => void;
-      afterSetCellMeta?: (row: number, col: number, key: string, value: any) => void;
+      afterSetCellMeta?: (row: number, column: number, key: string, value: any) => void;
       afterSetDataAtCell?: (changes: CellChange[], source?: ChangeSource) => void;
       afterSetDataAtRowProp?: (changes: CellChange[], source?: ChangeSource) => void;
       afterSetSourceDataAtCell?: (changes: CellChange[], source?: ChangeSource) => void;
-      afterTrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void;
+      afterTrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void | bolean;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1833]: bolean -> boolean

But anyway, `after*` hooks should not return anything.

File: handsontable.d.ts [hunk 2]
```diff
@@ -1826,17 +1826,17 @@ declare namespace Handsontable {
...
       afterUndo?: (action: plugins.UndoRedoAction) => void;
-      afterUnlisten?: () => void;
       afterUnhideColumns?: (currentHideConfig: number[], destinationHideConfig: number[], actionPossible: boolean, stateChanged: boolean) => void;
       afterUnhideRows?: (currentHideConfig: number[], destinationHideConfig: number[], actionPossible: boolean, stateChanged: boolean) => void;
+      afterUnlisten?: () => void;
       afterUnmergeCells?: (cellRange: wot.CellRange, auto: boolean) => void;
-      afterUntrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void;
+      afterUntrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void | boolean;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1839]: `after*` hooks should return `void`. Anything else will be ignored/

File: handsontable.d.ts [hunk 3]
```diff
@@ -1862,10 +1862,10 @@ declare namespace Handsontable {
       beforeDrawBorders?: (corners: number[], borderClassName: 'current' | 'area' | 'highlight' | undefined) => void;
       beforeDropdownMenuSetItems?: (menuItems: contextMenu.MenuItemConfig[]) => void;
       beforeDropdownMenuShow?: (instance: plugins.DropdownMenu) => void;
-      beforeFilter?: (conditionsStack: plugins.FiltersPlugin.ColumnConditions[]) => void | boolean;
+      beforeFilter?: (conditionsStack: plugins.FiltersPlugin.ColumnConditions[]) => boolean;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1865]: `beforeFilter` is not forcing boolean result, may not return anything. Keep `void`

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

**Comment by AMBudnik** [comment]
Hey @wojciechczerniak thank you for reviewing my PR. 
I was told to share it with @scarletfog but I've attached the changes so @scarletfog can work on the improved version.

File: handsontable.d.ts [hunk 4]
```diff
@@ -1826,15 +1826,15 @@ declare namespace Handsontable {
       afterSelectionByProp?: (row: number, prop: string, row2: number, prop2: string, preventScrolling: { value: boolean }, selectionLayerLevel: number) => void;
       afterSelectionEnd?: (row: number, column: number, row2: number, column2: number, selectionLayerLevel: number) => void;
       afterSelectionEndByProp?: (row: number, prop: string, row2: number, prop2: string, selectionLayerLevel: number) => void;
-      afterSetCellMeta?: (row: number, col: number, key: string, value: any) => void;
+      afterSetCellMeta?: (row: number, column: number, key: string, value: any) => void;
       afterSetDataAtCell?: (changes: CellChange[], source?: ChangeSource) => void;
       afterSetDataAtRowProp?: (changes: CellChange[], source?: ChangeSource) => void;
       afterSetSourceDataAtCell?: (changes: CellChange[], source?: ChangeSource) => void;
-      afterTrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void;
+      afterTrimRow?: (currentTrimConfig: number[], destinationTrimConfig: number[], actionPossible: boolean, stateChanged: boolean) => void | boolean;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1833]: One left

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: handsontable.d.ts (hunk 4 above)
Comment (AMBudnik) [reply handsontable.d.ts]: done

**Review by wojciechczerniak — DISMISSED** [review]
🏆

**Comment by scarletfog** [comment]
@wojciechczerniak just a small change within b44ff0fd098a9372480a32218ff31dd6ea5db83b after our discussion on slack.

**Review by wojciechczerniak — APPROVED** [review]
@scarletfog good catch 👍

File: handsontable.d.ts [hunk 5]
```diff
@@ -1765,16 +1765,16 @@ declare namespace Handsontable {
       afterCellMetaReset?: () => void;
       afterChange?: (changes: CellChange[] | null, source: ChangeSource) => void;
       afterChangesObserved?: () => void;
-      afterColumnCollapse?: (currentCollapsedColumn: number[], destinationCollapsedColumns: number[], collapsePossible: boolean, successfullyCollapsed: boolean) => void;
-      afterColumnExpand?: (currentCollapsedColumn: number[], destinationCollapsedColumns: number[], expandPossible: boolean, successfullyExpanded: boolean) => void;
-      afterColumnMove?: (columns: number[], target: number) => void;
+      afterColumnCollapse?: (currentCollapsedColumns: number[], destinationCollapsedColumns: number[], collapsePossible: boolean, successfullyCollapsed: boolean) => void;
+      afterColumnExpand?: (currentCollapsedColumns: number[], destinationCollapsedColumns: number[], expandPossible: boolean, successfullyExpanded: boolean) => void;
+      afterColumnMove?: (movedColumns: number[], finalIndex: number, dropIndex: number | undefined, movePossible: boolean, orderChanged: boolean) => void;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1770]: `afterRowMove` is using `void`

File: handsontable.d.ts [hunk 6]
```diff
@@ -1849,7 +1849,7 @@ declare namespace Handsontable {
       beforeChangeRender?: (changes: CellChange[], source: ChangeSource) => void;
       beforeColumnCollapse?: (currentCollapsedColumn: number[], destinationCollapsedColumns: number[], collapsePossible: boolean) => void | boolean;
       beforeColumnExpand?: (currentCollapsedColumn: number[], destinationCollapsedColumns: number[], expandPossible: boolean) => void | boolean;
-      beforeColumnMove?: (columns: number[], target: number) => void | boolean;
+      beforeColumnMove?: (movedColumns: number[], finalIndex: number, dropIndex: number | undefined, movePossible: boolean) => void | boolean;
```
Comment (wojciechczerniak) [inline handsontable.d.ts:1852]: `beforeRowMove` is using `void`

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

**Review by wojciechczerniak — APPROVED** [review]
🍪

**Pull request merged**